receive_buffer.go

```diff
@@ -78,8 +86,8 @@ func (buf *receiveBuffer) read(b []byte, deadline time.Time) (totalN int, err er
 		select {
 		case frame, open := <-buf.in:
 			// Read next frame, continue loop
-			if !open && frame == nil {
```

Comment:
Checking for frame == nil is unnecessary, if the channel is closed we won't have read anything from it.